ee/backend/app/src/main/kotlin/io/tolgee/ee/component/TaskNotificationEnhancer.kt

```diff
@@ -0,0 +1,25 @@
+package io.tolgee.ee.component
+
+import io.tolgee.ee.api.v2.hateoas.assemblers.TaskModelAssembler
+import io.tolgee.ee.service.TaskService
+import io.tolgee.hateoas.notification.NotificationEnhancer
+import io.tolgee.hateoas.notification.NotificationModel
+import io.tolgee.model.Notification
+import org.springframework.stereotype.Component
+
+@Component
+class TaskNotificationEnhancer(
+  private val taskService: TaskService,
+  private val taskModelAssembler: TaskModelAssembler,
+) : NotificationEnhancer {
+  override fun enhanceNotifications(notifications: Map<Notification, NotificationModel>) {
+    val taskIds = notifications.mapNotNull { (source, _) -> source.linkedTask?.id }
+    val convertedTasks = taskService.getTasksWithScope(taskIds)
```

Comment:
it wold be better to use .associateBy{it.id} on it and then to just get it from the map instead of searching in it using fing.

Reply:
Btw. Task have Id, so you should use it. Don't use the project id and number, because it will require you fetch it or it will be n+1.

Reply:
Hibernate log looks reasonable also, no `n+1`'s.